File: GamesDB/GamesDB/Utilizador.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GamesDB
{
    class Utilizador
    {
        private String _nome;
        private String _id;
        private String _telemovel;
        private String _email;
        private String _regiao;

        public string nome
        {
            get { return _nome; }
            set { _nome = value; }
        }

        public string id
        {
            get { return _id; }
            set { _id = value; }
        }

        public string telemovel
        {
            get { return _telemovel; }
            set { _telemovel = value; }
        }

        public string email
        {
            get { return _email; }
            set { _email = value; }
        }

        public string regiao
        {
            get { return _regiao; }
            set { _regiao = value; }
        }

        public override string ToString()
        {
            return String.Format("{0,-30}\t{1,-30}\t{2,-37}\t{3,-20}", _nome, _id, _telemovel, _email, _regiao);
        }

    }
}

File: GamesBD/GamesBD/Utilizador.vb
<Serializable()> Public Class Utilizador
    Private _Telemovel As String
    Private _Regiao As String
    Private _ID As String
    Private _Email As String
    Private _Nome As String

    Property Telemovel() As String
        Get
            Return _Telemovel
        End Get
        Set(value As String)
            If value Is Nothing Or value = "" Then
                Throw New Exception("Campo Telemóvel não pode estar vazio")
                Exit Property
            End If
            _Telemovel = value
        End Set
    End Property

    Property Regiao() As String
        Get
            Return _Regiao
        End Get
        Set(value As String)
            _Regiao = value
        End Set
    End Property

    Property ID() As String
        Get
            Return _ID
        End Get
        Set(value As String)
            _ID = value
        End Set
    End Property

    Property Email() As String
        Get
            Return _Email
        End Get
        Set(value As String)
            If value Is Nothing Or value = "" Then
                Throw New Exception("Campo Email não pode estar vazio")
                Exit Property
            End If
            _Email = value
        End Set
    End Property

    Property Nome() As String
        Get
            Return _Nome
        End Get
        Set(value As String)
            If value Is Nothing Or value = "" Then
                Throw New Exception("Campo Nome não pode estar vazio")
                Exit Property
            End If
            _Nome = value
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return _Nome & "  " & _Email & "  " & _Regiao & "  " & _ID
    End Function

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(ByVal Nome As String)
        MyBase.New()
        Me.Nome = Nome
    End Sub
End Class
